src/ipips/ipip-0002.md

```diff
@@ -0,0 +1,88 @@
+using a `_redirects` file hosted at the top level of the web site. While we do not intend to provide all of the same functionality, it seems desirable to use a similar approach to provide a meaningful subset of the functionality offered by these services.
+
+- The format is simple and low on syntax
+- Many developers are already familiar with this file name and format
+- Using a text file for configuration enables developers to make changes without using other IPFS tools
+- The configuration can be easily versioned in both version control systems and IPFS by virtue of the resulting change to the root CID for the content
+
+### User benefit
+
+Provides general URL redirect and rewrite support, which enables three important features:
+1. Developers will be able to host single-page applications in IPFS.
+2. Same configuration file used for setting up pretty 404 pages.
+3. The cost of switching hosting of an existing website to IPFS is lowered by making it possible to keep all legacy URLs working.
+
+### Compatibility
+
+If by some chance developers are already hosting sites that contain a `_redirects` file that does something else, they may need to update the contents of the file to match the new functionality. Errors returned to the user due to parsing errors will guide them regarding the required updates.
+
+### Alternatives
+
+- There was some discussion early on about a [manifest file](https://github.com/ipfs/specs/issues/257) that could be used to configure redirect support in addition to many other things. While the idea of a manifest file has merit, manifest files are much larger in scope and it became challenging to reach agreement on functionality to include.
+There is already a large need for redirect support for SPAs, and this proposal allows us to provide that critical functionality without being hampered by further design discussion around manifest files.
+In addition, similar to how Netlify allows redirect support to be configured in either a `_redirects` file or a more general [configuration file](https://docs.netlify.com/configure-builds/file-based-configuration/#redirects), there is nothing precluding IPFS from allowing developers to configure redirect support in an app manifest later on.
+- There was some discussion with the [n0](https://github.com/n0-computer/) team about potential ways to improve the performance of retrieving metadata such as redirect rules, possibly including it as metadata with the root CID such that it would be included with the request for the CID to begin with.
+I believe the performance concerns are alleviated by not providing forced redirect support, and looking for `_redirects` only if the DAG is missing a requested path.  Never the less, if a more generic metadata facility were to be introduced in the future, it may make sense to reconsider how redirect rules are specified.
+
+### Copyright
+
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```

Comment:
Was there a legal review on picking CC0? I ask because I know that the WHATWG switched from CC0 to CCBY4. I forget what the reason was but I can look it up.

Reply:
cc @lidel

Reply:
@darobin there was no legal review, intention was to "lock specs open" in the spirit of  https://protocol.ai/blog/announcing-the-permissive-license-stack/
if you could find out the reason for avoiding CC0 that would be really helpful (better to switch sooner than later)

Reply:
I don't think that this should block this update — I'll make a separate pass on doc license in genera.